### ISSUE TITLE
4.9 (13.b) nonstatic discriminant in aggregate

### ISSUE THREAD

**Issue by nholsti** [pat-rogers/Ada-202x-WG9-Informal-Review#167]
In this "Ramification", the last case, "when specifying the value of a discriminant ...", is no longer exactly true, because a nonstatic value is now allowed if the value has a static subtype that governs the variant part, RM 4.3.1(17/5).

**Comment by ARG-Editor** [comment]
You are correct. Moreover, this note reads as if this is a complete list of places where static expressions are required. But there has been no maintenance of this list in Ada 2005, Ada 2012, or Ada 202x, and it is almost certainly missing some constructs. It would be better to reduce it to a few interesting cases as examples, because no one is ever going to remember to maintain a list like this, (Rather, the complete list should be in the index as "static expression, required", but that's another can of worms at this late date - trying to find all of the places would be a pain.)

Anyway, I replaced the entire note with:

The language requires a static expression in a number_declaration, a numeric type definition, certain representation items, and a number of other places.

If someone would like to develop a complete list of current places, I'd be happy to put index entries in for them all. In that case, I'd add to the above " See 'static expression, required' in the index for a complete list of places".

**Comment by sttaft** [comment]
I would defer the new index entry to the next revision.  I would change "places" to "contexts" in your new wording.

**Comment by ARG-Editor** [comment]
I've marked this "deferred" so we remember to add the missing index entries.
